internal/server/iigointernal/executive.go

```diff
@@ -105,54 +134,60 @@ func (e *executive) replyAllocationRequest(commonPool shared.Resources, aliveIsl
+	if !e.incurServiceCharge(actionCost.AppointNextSpeakerActionCost) {
+		return e.ID, errors.Errorf("Insufficient Budget in common Pool: appointNextSpeaker")
+	}
 	var election voting.Election
 	election.ProposeElection(baseclient.Speaker, voting.Plurality)
 	election.OpenBallot(clientIDs)
 	election.Vote(iigoClients)
-	return election.CloseBallot()
-}
-
-// withdrawSpeakerSalary withdraws the salary for speaker from the common pool.
-func (e *executive) withdrawSpeakerSalary(gameState *gamestate.GameState) bool {
-	var speakerSalary = shared.Resources(rules.VariableMap[rules.SpeakerSalary].Values[0])
-	var withdrawnAmount, withdrawSuccesful = WithdrawFromCommonPool(speakerSalary, gameState)
-	e.speakerSalary = withdrawnAmount
-	return withdrawSuccesful
+	return election.CloseBallot(), nil
 }
 
-// sendSpeakerSalary send speaker's salary to the speaker.
-func (e *executive) sendSpeakerSalary(legislativeBranch *legislature) {
+// sendSpeakerSalary conduct the transaction based on amount from client implementation
+func (e *executive) sendSpeakerSalary() error {
 	if e.clientPresident != nil {
-		amountReturn := e.clientPresident.PaySpeaker(e.speakerSalary)
+		amountReturn := e.clientPresident.PaySpeaker()
 		if amountReturn.ActionTaken && amountReturn.ContentType == shared.PresidentSpeakerSalary {
-			legislativeBranch.budget = amountReturn.SpeakerSalary
+			// Subtract from common resources pool
+			amountWithdraw, withdrawSuccess := WithdrawFromCommonPool(amountReturn.SpeakerSalary, e.gameState)
+
+			if withdrawSuccess {
+				// Pay into the client private resources pool
+				depositIntoClientPrivatePool(amountWithdraw, e.gameState.SpeakerID, e.gameState)
+				return nil
+			}
 		}
-		return
 	}
-	legislativeBranch.budget = e.speakerSalary
+	return errors.Errorf("Cannot perform sendJudgeSalary")
 }
 
 func (e *executive) reset(val string) {
 	e.ID = 0
 	e.clientPresident = nil
-	e.budget = 0
 	e.ResourceRequests = map[shared.ClientID]shared.Resources{}
 	e.RulesProposals = []string{}
 	e.speakerSalary = 0
 }
 
-//requestRuleProposal asks each island alive for its rule proposal.
-func (e *executive) requestRuleProposal(aliveIslands []shared.ClientID) {
-	e.budget -= serviceCharge
-	var rules []string
-	for _, islandID := range aliveIslands {
-		rules = append(rules, iigoClients[islandID].RuleProposal())
-	}
+// Helper functions:
+func (e *executive) getTaxMap(islandsResources map[shared.ClientID]shared.ResourcesReport) shared.PresidentReturnContent {
+	return e.clientPresident.SetTaxationAmount(islandsResources)
+}
 
-	e.setRuleProposals(rules)
+// incur charges in both budget and commonpool for performing an actions
+// actionID is typically the function name of the action
+// only return error if we can't withdraw from common pool
+func (e *executive) incurServiceCharge(cost shared.Resources) bool {
+	_, ok := WithdrawFromCommonPool(cost, e.gameState)
```

Comment:
What is the ignored return value?

Reply:
It's just the value of cost

Reply:
Yeah, we don't need to know how much it returned as it will return the full amount we put it. Not sure why it returns two values, but I guess someone has a use for it